vllm/entrypoints/llm.py

```diff
@@ -152,7 +153,7 @@ def set_tokenizer(
         self,
         tokenizer: Union[PreTrainedTokenizer, PreTrainedTokenizerFast],
     ) -> None:
-        self.llm_engine.tokenizer.tokenizer = tokenizer
```

Comment:
better add a check, if the tokenizer is already cached, then just set it.

Reply:
A check is added. While the CachedTokenizer is dynamic,  I have no choice but compare the class name. Is there any good ideal?